docs/cloud/rahti/gpu.md

```diff
@@ -0,0 +1,103 @@
+
+```yaml
+# gpu-example-1.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: gpu-example-1
+spec:
+  restartPolicy: Never
+  containers:
+    - name: gpu-example-1
+      image: "nvidia/cuda:11.6.0-base-ubi8"
+      command: ["nvidia-smi"]
+      resources:
+        limits:
+          nvidia.com/gpu: 1
+```
+
+And run the follwing command to start the pod: 
+```bash
+oc apply -f gpu-example-1.yaml
+```
+
+Finally check the output of the execution from the pod (you might need to wait for a few seconds for the pod creation to complete) and clean up after using: 
+```bash
+oc logs gpu-example-1
+```
+```bash
+oc delete pod gpu-example-1
+```
+
+As second example you can create a pod running Cuda based vector additon to demonstrate access to GPU through a lower-level programming model/interface. 
+
+To run the second example create a pod defination file as: 
+
+```yaml
+# gpu-example-2.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: gpu-example-2
+spec:
+  restartPolicy: OnFailure
+  containers:
+    - name: gpu-example-2
+      # https://github.com/kubernetes/kubernetes/blob/v1.7.11/test/images/nvidia-cuda/Dockerfile
+      image: "k8s.gcr.io/cuda-vector-add:v0.1"
+      resources:
+        limits:
+          nvidia.com/gpu: 1 # requesting 1 GPU
+```
+
+And run, check the log (you might need to wait for a few seconds for the task to complete), and clean up after as:
+
+```bash
+oc apply -f gpu-example-2.yaml
+```
+```bash
+oc logs gpu-example-2
+```
+```bash
+oc delete pod gpu-example-2
+```
+
+As a final example, we show how a more complete system that uses GPU resources can be deployed on Rahti. The example provides an API for training and deploying a text sentiment classification model. To facilitate easier deployment, we use various Kubernetes resource definitions and their explanation can be found in the general [Kubernetes basics page](https://docs.csc.fi/cloud/rahti/concepts/).  One of the main things to notice in this example is the Dockerfile that packages the application. The Dockerfile uses an appropriate base image (`FROM tensorflow/tensorflow:2.7.0-gpu`) which contains the right runtime CUDA libraries for utilizing the GPU resource. This image can be built locally and be pushed to an image repository that can be used during the deployment of the system. A`BuildConfig` Openshift resource can also be used as is the case in this example. 
+
+To run the example use the following command: 
+```bash
+oc process -f https://raw.githubusercontent.com/tdeneke/rahti-ml-examples/tf2-imdb-cuda/rahti-ml-example-cuda-template.yaml | oc create -f -
+```
+
+You can then follow the deployment of the application and its components from the Rathi console or using `oc status`, and interact with the application using the UI at <your-application-route>/docs. The route your application has been deployed to can be found from the Rahti console or using `oc get routes`.  
+
+After exploring and testing the example deployment it can be cleaned up as:
```

Comment:
Use imperative mode instead of conditional: it can be cleaned up -> it should be cleaned up

(otherwise new users will assume that cleaning up is optional, which will eventually lead to all GPUs being reserved)